File: SemTumultoApi/Models/Empresas/Empresa.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using SemTumultoApi.Models.Reservas;
using SemTumultoApi.Models.Usuarios;

namespace SemTumultoApi.Models.Empresas
{
    public class Empresa
    {
        public Guid Id { get; set; }
        public Guid UsuarioId { get; set; }

        [Required(ErrorMessage = "O campo Nome é obrigatório")]
        public string Nome { get; set; }

        [Required(ErrorMessage = "O campo Descrição é obrigatório")]
        public string Descricao { get; set; }
        public List<Reserva> Reservas { get; set; }
        public Usuario Usuario { get; set; }

    }
}

File: front/src/app/models/empresa/empresa.ts
import { Usuario } from '../usuario/usuario';
import { Reserva } from '../reserva/reserva';
import { Data } from '@angular/router';

export interface Empresa {
  id: string;
  usuarioId: string;
  nome: string;
  descricao: string;
  reservas: [
    {
      id: string;
      usuarioId: string;
      usuario: Usuario[];
      empresaId: string;
      concluida: boolean;
      dataHoraReserva: Date;
    }
  ];
  usuario: Usuario[];
}
